pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -101,6 +101,11 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation) *gardencorev1
 			Name: "Deploying Shoot namespace in Seed",
 			Fn:   flow.TaskFn(botanist.DeployNamespace).RetryUntilTimeout(defaultInterval, defaultTimeout),
 		})
+		ensureShootClusterIdentity = g.Add(flow.Task{
+			Name:         "Ensuring Shoot cluster identity in Shoot's namespace in the Seed",
+			Fn:           flow.TaskFn(botanist.EnsureClusterIdentity).RetryUntilTimeout(defaultInterval, defaultTimeout),
```

Comment:
Gardener gradually moves towards testable components in the botanist e.g. for the [Network CRD deployment](https://github.com/gardener/gardener/pull/2442). I think new reconciliation steps should be implemented with components and unit tests.
I know this will result in higher initial effort, but it makes sense in the long run.